### PR TITLE
Cleanup and PSH route minor adjustments

### PR DESCRIPTION
**Pull request by eanders**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

This:
1. Centralizes the determination of if a step is "editable?"
2. Hides the skipped steps on routes where hearings are ignored.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)
- [x] Code clean-up / housekeeping

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [ ] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [ ] My code follows the style guidelines of this project (rubocop)
- [ ] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Comment by eanders** [comment]
When a sub-program does not require a CORI check
Screenshot 2024-08-29 at 4 15 01 PM

Hide two steps from the UI
Screenshot 2024-08-29 at 4 14 53 PM

When a CORI check is required
Screenshot 2024-08-29 at 4 15 09 PM

Show the steps
Screenshot 2024-08-29 at 4 15 30 PM

File: app/models/match_decisions/base.rb [hunk 1]
```diff
@@ -86,7 +86,7 @@ def contact_name
     def editable?
       # can this decision be updated by a notification response?
       # override this default behavior in subclasses
-      initialized? && match_open?
+      initialized? && match_open? && saved_status !~ /acknowledged|accepted|confirmed|declined|canceled|rejected|complete|completed|scheduled|no_hearing|mitigation_required|mitigation_not_required|decline_overridden|decline_overridden_returned|decline_confirmed/
```
Comment (ttoomey) [inline app/models/match_decisions/base.rb]: does this regex want anchors? Assuming we always want to match the entire status 
```suggestion
      initialized? && match_open? && saved_status !~ /\A(acknowledged|accepted|confirmed|declined|canceled|rejected|complete|completed|scheduled|no_hearing|mitigation_required|mitigation_not_required|decline_overridden|decline_overridden_returned|decline_confirmed)\z/
```

File: app/views/match_list_base/_current_step.haml [hunk 2]
```diff
@@ -1,6 +1,12 @@
 - match_route = match.match_route
 - types = match_step_types[match_route.class.name]
 - match_decisions = match.decisions.order(id: :asc).where(type: types&.keys).to_a
+- # remove any skipped decisions
+- skipped_names = match_decisions.select(&:skipped?).map { |m| m.class.name }
+- match_decisions.reject!(&:skipped?)
+- types.reject! { |class_name, _| class_name.in?(skipped_names) }
+- i = 0
+- types.transform_values! { |v| i += 1 }
```
Comment (ttoomey) [inline app/views/match_list_base/_current_step.haml]: maybe could be golfed down to
```suggestion
- types = types.keys.zip(1..).to_h
```

**Review by ttoomey — APPROVED** [review]
this is great, thanks!

**Pull request merged**